flowmachine/flowmachine/features/subscriber/distance_series.py

```diff
@@ -0,0 +1,140 @@
+        statistic: str = "avg",
+        time_bucket: str = "day",
+    ):
+        subscriber_locations.spatial_unit.verify_criterion("has_geography")
+        subscriber_locations.spatial_unit.verify_criterion("has_lon_lat_columns")
+        self.spatial_unit = subscriber_locations.spatial_unit
+        if time_bucket.lower() in valid_time_buckets:
+            self.aggregate_by = time_bucket.lower()
+        else:
+            raise ValueError(
+                f"'{time_bucket}' is not a valid value for time_bucket. Use one of {valid_time_buckets}"
+            )
+
+        self.statistic = statistic.lower()
+        if self.statistic not in valid_stats:
+            raise ValueError(
+                "{} is not a valid statistic. Use one of {}".format(
+                    self.statistic, valid_stats
+                )
+            )
+        self.start = subscriber_locations.start
+        self.stop = subscriber_locations.stop
+        if reference_location is None:
+            # Using 0, 0
+            self.reference_location = None
+            self.joined = subscriber_locations
+        elif isinstance(reference_location, BaseLocation):
+            if reference_location.spatial_unit != subscriber_locations.spatial_unit:
+                raise ValueError(
+                    "reference_location must have the same spatial unit as subscriber_locations."
+                )
+            self.reference_location = reference_location
+            self.joined = reference_location.join(
+                other=subscriber_locations,
+                on_left=["subscriber"],
+                left_append="_from",
+                right_append="_to",
+            ).join(
+                DistanceMatrix(spatial_unit=self.spatial_unit),
+                on_left=[
+                    f"{col}_{direction}"
+                    for direction in ("from", "to")
+                    for col in self.spatial_unit.location_id_columns
+                ],
+                right_append="_dist",
+                how="left outer",
+            )
+        else:
+            raise ValueError(
+                "Argument 'reference_location' should be an instance of BaseLocation class or None. "
+                f"Got: {type(reference_location)}"
+            )
+
+        super().__init__()
+
+    @property
+    def column_names(self) -> List[str]:
+        return ["subscriber", "datetime", "value"]
+
+    def _make_query(self):
+        if self.reference_location is None:
+            joined = f"""
+            SELECT subscriber, time as time_to, ST_Distance(ST_Point(0, 0)::geography, ST_Point(lon, lat)::geography) as value_dist
+            FROM ({self.joined.get_query()}) _
+            """
+        else:
+            joined = self.joined.get_query()
+
+        if valid_time_buckets.index(self.aggregate_by) > valid_time_buckets.index(
+            "hour"
+        ):  # Slightly nicer to cast things which aren't timestamps to a date
```

Comment:
I can definitely see the convenience in this. I'm slightly in two minds about it because it makes the return type inconsistent (which may make further processing awkward if it's not done manually), but that's probably ok. We should mention the automated casting in the docstring, though.